sdk/messaging/azservicebus/admin/admin_client_rules.go

```diff
@@ -0,0 +1,558 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package admin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus/internal/atom"
+)
+
+// Filters allows you to specify a single filter for a particular rule.
+// Only one of these properties can be set.
```

Comment:
Is this how other languages model the filters?  It seems a little clumsy to have a type with all of these mutually exclusive fields.  Can be refined in a separate PR as necessary.

Reply:
The other languages basically have the equivalent of an 'any' field for the filter, which I believe we could also do. So the user would have to do a casting but that would work. 

What do you think of that?

Reply:
(also, for better or for worse, I modeled it after this: https://github.com/Azure/azure-sdk-for-go/blob/8d5dc19fa7e427d90690ecc116feb8159c9e9c45/services/servicebus/mgmt/2017-04-01/servicebus/models.go#L2176)

Reply:
Let's schedule some time with @JeffreyRichter to come up with a design.  I assume that there's no urgency in releasing this and we can clean it up in a separate PR later.

Reply:
I changed it to any 'any' field for now anyways. We can work on it more but this is at least inline with what other languages do.